File: zema_store_mobile/lib/models/role.dart
class Role {
  Role({
    required this.name,
    required this.createdAt,
    required this.updatedAt,
    });


  final String name;
  final DateTime createdAt;
  final DateTime updatedAt;

  @override
  List<Object> get props => [ name,createdAt,updatedAt];

  factory Role.fromJson(Map<String, dynamic> json) {
    
    return Role(
      name: json['name'],
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt']
    );
  }
}

File: zema_store_backend/src/models/mongoose/role.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IRoleDocument extends Document {
  name: String;
  createdAt: Date;
  updatedAt: Date;
}

const roleSchema: Schema<IRoleDocument> = new Schema(
  {
    name: {
      type: String,
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

const Role = mongoose.model<IRoleDocument>("Role", roleSchema);

roleSchema.virtual("id").get(function (this: IRoleDocument) {
  return this._id.toHexString();
})

export default Role;
